tern.js

```diff
@@ -38,11 +38,72 @@ define(function(require, exports, module) {
                 // TODO: register "extra" defs?
             }
             */
-
-            registerDef("jQuery", "lib/tern/defs/jquery.json", true),
-            registerDef("Browser built-in", "lib/tern/defs/browser.json", true);
-            registerDef("Underscore", "lib/tern/defs/underscore.json"),
-            registerDef("Chai", "tern/defs/chai.json");
+            
+            var ternOptions = options.tern || {
+                plugins: {
+                    doc_comment: "tern/plugin/doc_comment",
+                    smartface: "plugins/smartface/loadInclude.js",
+                    angular: "tern/plugin/angular",
+                    component: "tern/plugin/component",
+                    doc_comment: "tern/plugin/doc_comment",
+                    node: "tern/plugin/node",
+                    requirejs: "tern/plugin/requirejs",
+                    architect_resolver: "./architect_resolver_worker"
+                },
+                defs: [
+                    {
+                        name: "ecma5",
+                        enabled: true,
+                        path: "lib/tern/defs/ecma5.json"
+                    },
+                    {
+                        name: "jQuery",
+                        enabled: true,
+                        path: "lib/tern/defs/jquery.json"
+                    },
+                    {
+                        name: "browser",
+                        enabled: true,
+                        path: "lib/tern/defs/browser.json"
+                    },
+                    {
+                        name: "underscore",
+                        enabled: false,
+                        path: "lib/tern/defs/underscore.json"
+                    },
+                    {
+                        name: "chai",
+                        enabled: false,
+                        path: "tern/defs/chai.json"
+                    }
+                ]
+            };
+            
+            ternPlugins(function callback(e) {
+                var pluginName, pluginPath;
+                for(pluginName in ternOptions.plugins) {
+                    pluginPath = ternOptions.plugins[pluginName];
+                    e.push({name: pluginName, enabled: true, path: pluginPath});    
+                }
+            });
+            
+            
+            var defsToAdd =[], defIndex, d;
+            for(defIndex in ternOptions.defs) {
```

Comment:
Sorry to bug you about this, but please check your PR against our [coding standards](https://github.com/c9/core/blob/master/docs/CODING_STANDARDS.md), particularly looking for:
- space after `=` and keywords like `for` and `if`
- declarations on separate lines (this is perhaps a soft rule, but it's especially good to do this when initializing in declarations)
- use double quotes `"` for string literals